### DISCUSSION TITLE
Dynamic XML badge with HTML page returns "invalid"

### DISCUSSION THREAD

**Discussion by pke** [badges/shields#9701] [Q&A]
I tried to parse a [HTML5 page](https://greasyfork.org/en/scripts/406337-toc-bar-auto-generating-table-of-content) with a custom XML badge but the resulting badge just shows "invalid".  I am using this XPath: `//html/head/title`, which should yield the title.
I assume that has to do with the HTML5 `<!DOCTYPE` prefix, that the XML parser chokes on?
Last resort would be to implement a endpoint badge but maybe we could also add a "ignore HTML5 DOCTYPE option" to the XML parser?

When I am using this test script in node (from the xml service here), it can parse the XPath just fine

```ts
import {DOMParser} from "@xmldom/xmldom"
import xpath from "xpath"

const pathExpression = "//head/title/text()"
const parser = new DOMParser()
fetch("https://greasyfork.org/en/scripts/406337-toc-bar-auto-generating-table-of-content")
.then(result => result.text())
.then(html => parser.parseFromString(html))
.then(html => {
  const pathIsAttr = (
    pathExpression.split('/').slice(-1)[0] || ''
  ).startsWith('@')

  let values
  try {
    values = xpath.select(pathExpression, html)
  } catch (e) {
    throw new InvalidParameter({ prettyMessage: e.message })
  }
  if (
    typeof values === 'string' ||
    typeof values === 'number' ||
    typeof values === 'boolean'
  ) {
    values = [values]
  } else if (Array.isArray(values)) {
    values = values.reduce((accum, node) => {
      if (pathIsAttr) {
        accum.push(node.value)
      } else if (node.firstChild) {
        accum.push(node.firstChild.data)
      } else {
        accum.push(node.data)
      }

      return accum
    }, [])
  } else {
    throw new InvalidResponse({
      prettyMessage: 'unsupported query',
    })
  }

  if (!values.length) {
    throw new InvalidResponse({ prettyMessage: 'no result' })
  }

  return { values }
})
.catch(error => console.log("error", error))
```

I also tried the JSON dynamic badge with this query:
`$..dd[?(@['@class']=='script-show-total-installs')].span`

```json
[
  "5,961"
]
```

which works on https://jsonpath.com/ when the websites source code is pasted in there.

But the custom badge query:

`![Dynamic JSON Badge](https://img.shields.io/badge/dynamic/json?url=https%3A%2F%2Fgreasyfork.org%2Fen%2Fscripts%2F406337-toc-bar-auto-generating-table-of-content&query=%24..dd%5B%3F(%40%5B'%40class'%5D%3D%3D'script-show-total-installs')%5D.span)`

returns this ![Dynamic JSON Badge](https://img.shields.io/badge/dynamic/json?url=https%3A%2F%2Fgreasyfork.org%2Fen%2Fscripts%2F406337-toc-bar-auto-generating-table-of-content&query=%24..dd%5B%3F(%40%5B'%40class'%5D%3D%3D'script-show-total-installs')%5D.span)

**Comment by PyvesB** [comment]
@pke are you still having trouble with this use case?

**Reply by pke** [reply]
Yes, the problem still exists. Maybe my markdown markup is not correct?
